src/preprocessing/passes/bv_to_int.cpp

```diff
@@ -0,0 +1,981 @@
+
+  while (!toVisit.empty())
+  {
+    Node current = toVisit.back();
+    uint64_t currentNumChildren = current.getNumChildren();
+    if (d_bvToIntCache.find(current) == d_bvToIntCache.end())
+    {
+      // This is the first time we visit this node and it is not in the cache.
+      d_bvToIntCache[current] = Node();
+      toVisit.insert(toVisit.end(), current.begin(), current.end());
+    }
+    else
+    {
+      // We already visited this node
+      if (!d_bvToIntCache[current].isNull())
+      {
+        // We are done computing the translation for current
+        toVisit.pop_back();
+      }
+      else
+      {
+        // We are now visiting current on the way back up.
+        // This is when we do the actual translation.
+        if (currentNumChildren == 0)
+        {
+          if (current.isVar())
+          {
+            if (current.getType().isBitVector())
+            {
+              // For bit-vector variables, we create integer variables and add a
+              // range constraint.
+              Node newVar = d_nm->mkSkolem("__bvToInt_var",
+                                           d_nm->integerType(),
+                                           "Variable introduced in bvToInt "
+                                           "pass instead of original variable "
+                                               + current.toString());
+
+              d_bvToIntCache[current] = newVar;
+              d_rangeAssertions.insert(mkRangeConstraint(
+                  newVar, current.getType().getBitVectorSize()));
+            }
+            else
+            {
+              // Boolean variables are left unchanged.
+              AlwaysAssert(current.getType() == d_nm->booleanType()
+                           || current.getType().isSort());
+              d_bvToIntCache[current] = current;
+            }
+          }
+          else if (current.isConst())
+          {
+            if (current.getKind() == kind::CONST_BITVECTOR)
+            {
+              // Bit-vector constants are transformed into their integer value.
+              BitVector constant(current.getConst<BitVector>());
+              Integer c = constant.toInteger();
+              d_bvToIntCache[current] = d_nm->mkConst<Rational>(c);
+            }
+            else
+            {
+              // Other constants stay the same.
+              d_bvToIntCache[current] = current;
+            }
+          }
+          else
+          {
+            Unimplemented();
+          }
+        }
+        else
+        {
+          /**
+           * The current node has children.
+           * Since we are on the way back up,
+           * these children were already translated.
+           * We save their translation for future use.
+           */
+          vector<Node> intized_children;
+          for (uint64_t i = 0; i < currentNumChildren; i++)
+          {
+            intized_children.push_back(d_bvToIntCache[current[i]]);
```

Comment:
I personally find `intized_children` to be a bit of a weird name choice, made me double back while reading through the code and was not immediately obvious what it is supposed to mean. Maybe use a better name?

Reply:
I find-replaced it to `translated_children`.